### ISSUE TITLE
[Crash] stack overflow on call to Landroid/graphics/Paint;.getTextRunAdvance:FLIIIIILI 

### ISSUE THREAD

**Issue by InnerPeace080** [facebook/react-native#5404]
i run my program with react-native version 0.17.0 , and everything seem normal.
But when i upgrade to 0.18.0 , my program crash with following error 

stack overflow on call to Landroid/graphics/Paint;.getTextRunAdvance:FLIIIIILI

seem like something change in react-native version 0.18.0 make it crash. how can i slove this?

**Comment by MITDelian** [comment]
I've also experienced this

**Comment by MITDelian** [comment]
This appears to be a known bug with TextView on Android: http://stackoverflow.com/questions/12849676/i-cant-figure-the-issue-with-a-stacktrace-error

**Comment by MITDelian** [comment]
Hmm there appears to be some sort of maximum depth to the view hierarchy on android. Coming up with a simple use case.

**Comment by MITDelian** [comment]
Ok so I made a pretty simple use case out of this. I started with v0.20.0 and used react-native init and all I did was change the index.android.js to the following

```
/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 */
'use strict';
import React, {
  AppRegistry,
  Component,
  StyleSheet,
  Text,
  View
} from 'react-native';

class AwesomeProject extends Component {
  render() {
    return (
      <View style={styles.container}>
        <Text style={styles.welcome}>
          Welcome to React Native!
        </Text>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
        <View>
          <Text style={styles.instructions}>
            To get started, edit index.android.js
          </Text>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        </View>
        <Text style={styles.instructions}>
          Shake or press menu button for dev menu
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  welcome: {
    fontSize: 20,
    textAlign: 'center',
    margin: 10,
  },
  instructions: {
    textAlign: 'center',
    color: '#333333',
    marginBottom: 5,
  },
});

AppRegistry.registerComponent('AwesomeProject', () => AwesomeProject);
```

After ~16 levels it causes the error above and the app crashes. Note this is on a Samsung Galaxy Tab running Android 4. So far every device I have on hand that uses Android 4 crashes. Same thing with any Genymotion emulators on Android 4.

**Comment by MITDelian** [comment]
Looks like this is a limitation of the call stack on Android?

http://stackoverflow.com/questions/16913317/create-the-ui-thread-manually-in-android-in-order-to-increase-its-small-stack-si

http://stackoverflow.com/questions/16843357/what-is-the-android-ui-thread-stack-size-limit-and-how-to-overcome-it

**Comment by MITDelian** [comment]
Yeah I'm honestly unconvinced there is much of a fix? I basically just used the inspector and flattened my view structure as much as I could so that I didn't hit this upper limit.

Might just be useful including in the Android docs that there is a limitation to the view tree depth on Android 4 that is roughly ~20 views in total before you overflow the call stack on rendering.

**Comment by MITDelian** [comment]
@ide I don't think there's really a fix for this as I've dug deeper and deeper. It's a limitation on the call stack size in later versions of Android. Maybe somehow in the CSS layout android code you limit or throw a warning?

I'd be happy to also just submit documentation on this so that others aren't caught unawares

**Comment by ide** [comment]
We could try boosting the stack size since this code runs on a thread other than the UI thread: http://stackoverflow.com/a/19967816. A better solution is to try rewriting the code to use BFS instead of DFS but anyone want to give the stack depth tweak a shot?

**Comment by ide** [comment]
I think this has been fixed with 0.27.

**Comment by MITDelian** [comment]
holy crap
